src/main/java/co/novu/common/rest/RestHandler.java

```diff
@@ -35,6 +36,14 @@ public Retrofit buildRetrofit() {
                             .build();
                     return chain.proceed(request);
                 }).addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BASIC));
+        
+        if(novuConfig.isEnableRetry()) {
+        	clientBuilder.addInterceptor(new RetryInterceptor(novuConfig.getMaxRetries(), novuConfig.getMinRetryDelayMillis() , novuConfig.getMaxRetryDelayMillis() , novuConfig.getInitialRetryDelayMillis()));
+        }
+        
+        if(novuConfig.isEnableIdempotencyKey()) {
+        	clientBuilder.addInterceptor(new IdempotencyKeyInterceptor());
+        }
```

Comment:
The main reason why this can't be done is the fact that we provide the Retrofit instance lazily (see lines 27 to 29). Doing this would mean that every request will use the same value as the `Idempotency-Key`, which is not the goal

Reply:
@mayorJAY I tried this interceptor and it is inputting new header value for each request made